GiEnJul/ClientApp/src/components/admin/overview/OverviewMacro.tsx

```diff
@@ -0,0 +1,191 @@
+import React, { useEffect, useState } from "react";
+import Giver from "./Giver";
+import Recipient from "./Recipient";
+import { Button, Grid, Typography } from "@material-ui/core";
+import usePost from "../../../hooks/usePost";
+
+export interface SelectedConnectionType {
+  giverRowKey: string;
+  giverPartitionKey: string;
+  recipientRowKey: string;
+  recipientPartitionKey: string;
+}
+
+export interface GiverType {
+  email: string;
+  eventName: string;
+  fullName: string;
+  hasConfirmedMatch: Boolean;
+  isSuggestedMatch: Boolean;
+  location: string;
+  matchedRecipient?: string;
+  maxReceivers: Number;
+  partitionKey: string;
+  rowKey: string;
+  phoneNumber: string;
+  isSelected: Boolean;
+}
+
+export interface RecipientType {
+  contactEmail: string;
+  contactFullName: string;
+  contactPhoneNumber: string;
+  dessert: string;
+  dinner: string;
+  eventName: string;
+  familyMembers: PersonType[];
+  hasConfirmedMatch: Boolean;
+  institution: string;
+  isSuggestedMatch: Boolean;
+  location: string;
+  matchedGiver?: GiverType;
+  note: string;
+  partitionKey: string;
+  referenceId: string;
+  rowKey: string;
+  isSelected: Boolean;
+}
+export interface PersonType {
+  partitionKey: string;
+  rowKey: string;
+  wish: string;
+  age: Number;
+  gender: Number;
+}
+
+function OverviewMacro() {
+  const [selectedConnection, setSelectedConnection] =
+    useState<SelectedConnectionType>({
+      giverRowKey: "",
+      giverPartitionKey: "",
+      recipientRowKey: "",
+      recipientPartitionKey: "",
+    });
+  const [giverData, setGiverData] = useState<GiverType[] | []>([]);
+  const [recipientData, setRecipientData] = useState<RecipientType[] | []>([]);
+
+  async function fetchGivers() {
```

Comment:
The `fetchGivers` and `fetchRecipients` should be changed into using the `useFetch` function instead.

Reply:
not working properly atm